returnn/config.py

```diff
@@ -724,9 +628,10 @@ def tf_should_use_gpu(config):
     if is_gpu_available():
       print("Device not set explicitly, and we found a GPU, which we will use.", file=log.v2)
       config.set("device", "gpu")
+      return True
     else:
       print("Device not set explicitly, and no GPU found.", file=log.v2)
       config.set("device", "cpu")
-  devs = get_devices_init_args(config)
-  assert len(devs) == 1, "multiple devices not supported yet for TF"
-  return any([d["device"].startswith("gpu") for d in devs])
+      return False
+  else:
+    raise AssertionError("Currently unsupported TF device %s specified" % cfg_dev)
```

Comment:
Why `AssertionError`? Either normal `Exception`, or maybe `ValueError`.
Also, use `%r`.
```suggestion
    raise ValueError("Currently unsupported TF device %r specified" % (cfg_dev,))
```